.github/workflows/reusable-test.yml

```diff
@@ -0,0 +1,76 @@
+---
+name: Reusable test
+
+permissions: {}
+
+on:
+  workflow_call:
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    name: Test Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      matrix:
```

Comment:
Don't put the matrix inside the reusable workflows, keep it in the top-level one and pass params as workflow inputs.